cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -168,6 +169,8 @@ func NewStaticAutoscaler(
 	ndt := deletiontracker.NewNodeDeletionTracker(0 * time.Second)
 	scaleDown := legacy.NewScaleDown(autoscalingContext, processors, ndt, deleteOptions)
 	actuator := actuation.NewActuator(autoscalingContext, clusterStateRegistry, ndt, deleteOptions)
+	autoscalingContext.ScaleDownActuator = actuator
```

Comment:
Hmm, it's really confusing to have this implementation in `AutoscalingContext`, and a different wrapped one in `StaticAutoscaler`, especially since they have the exact same name. Could we make `StaticAutoscaler` use the direct version from `AutoscalingContext` and wrap it internally/on demand somehow?

@x13n Do I understand correctly that after we switch to the parallel scale-down planner in 1.26 no wrapping would be needed since `StaticAutoscaler` could just use the direct actuator implementation from the context without any wrapping? If so, maybe we could just add a TODO to make this change after we switch the planner?

Reply:
Adding TODO makes sense. Yes, we'll be able to use actuator from context eventually, but not so quickly - in 1.26/1.27 both implementations will be available (one will be selected based on a feature flag).